crates/ruff_python_formatter/src/statement/stmt_try.rs

```diff
@@ -1,17 +1,120 @@
-use crate::{not_yet_implemented, FormatNodeRule, PyFormatter};
+use crate::comments;
+use crate::comments::leading_alternate_branch_comments;
+use crate::prelude::*;
+use crate::statement::FormatRefWithRule;
+use crate::{FormatNodeRule, PyFormatter};
 use ruff_formatter::{write, Buffer, FormatResult};
+use ruff_python_ast::node::AstNode;
+use ruff_python_ast::prelude::{ExceptHandler, Ranged};
 use rustpython_parser::ast::StmtTry;
 
 #[derive(Default)]
 pub struct FormatStmtTry;
 
+#[derive(Copy, Clone, Default)]
+pub struct FormatExceptHandler;
+
+impl FormatRule<ExceptHandler, PyFormatContext<'_>> for FormatExceptHandler {
+    fn fmt(
+        &self,
+        item: &ExceptHandler,
+        f: &mut Formatter<PyFormatContext<'_>>,
+    ) -> FormatResult<()> {
+        match item {
+            ExceptHandler::ExceptHandler(x) => x.format().fmt(f),
+        }
+    }
+}
+
+impl<'ast> AsFormat<PyFormatContext<'ast>> for ExceptHandler {
+    type Format<'a> = FormatRefWithRule<
+        'a,
+        ExceptHandler,
+        FormatExceptHandler,
+        PyFormatContext<'ast>,
+    > where Self: 'a;
+
+    fn format(&self) -> Self::Format<'_> {
+        FormatRefWithRule::new(self, FormatExceptHandler::default())
+    }
+}
+
 impl FormatNodeRule<StmtTry> for FormatStmtTry {
     fn fmt_fields(&self, item: &StmtTry, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let StmtTry {
+            range: _,
+            body,
+            handlers,
+            orelse,
+            finalbody,
+        } = item;
+
+        let comments_info = f.context().comments().clone();
+        let mut dangling_comments = comments_info.dangling_comments(item.as_any_node_ref());
+
+        write!(
+            f,
+            [
+                text("try:"),
+                hard_line_break(),
+                block_indent(&body.format()),
+            ]
+        )?;
+
+        let mut last_node = body.last();
+
+        for handler in handlers {
+            let handler_comments_start =
+                dangling_comments.partition_point(|comment| comment.slice().end() <= handler.end());
+            let (handler_comments, rest) = dangling_comments.split_at(handler_comments_start);
+            dangling_comments = rest;
+
+            write!(
+                f,
+                [
+                    leading_alternate_branch_comments(handler_comments, last_node),
+                    &handler.format()
+                ]
+            )?;
```

Comment:
Related to [the conversation about `leading_alternate_branch_comments`](https://github.com/astral-sh/ruff/pull/5222#discussion_r1238493849). 

```suggestion
           let handler_comments = comments_info.leading_comments(handler);
			write!(
			    f,
			    [
			        leading_alternate_branch_comments(handler_comments, last_node),
			        &handler.format()
			    ]
			)?;
```

Reply:
i don't think this is quite right. the first except handler seems to be working differently to subsequent ones.

i'm still debugging but it seems the comments just above the first except handler and the finally clause are handled by `handle_in_between_bodies_own_line_comment` whereas the ones above except handlers 2-n are handled by `handle_in_between_except_handlers_or_except_handler_and_else_or_finally_comment`

Reply:
ok. if i understand correctly i just need both, i.e. special handling of comments for the first except handler.

Reply:
Can you share an example? My understanding is that `handle_in_between_bodies_own_line_comment` should always attach the comment as leading comment and `handle_in_between_except_handlers_or_except_handler_and_else_or_finally_comment` should only attach `dangling_comments` for comments before the `finally` or `orelse` body.

Reply:
```
try:
    ...

except:
    ...

# before except 2
except:
    ...
```

this comment gets attached as dangling by `handle_in_between_except_handlers_or_except_handler_and_else_or_finally_comment`

Reply:
Okay. This is incorrect. This should become a leading comment. Feel free to leave a todo.

Reply:
ah, think i figured out how to fix it.